### PR TITLE
feat: add React18 to peerDeps

### LINKED ISSUE #99: Support React@18
**Describe the bug**
I'm currently using React 18 in my project and I'm having issues when I try to install this package because of the react version specified in https://github.com/storybookjs/testing-react/blob/d22cb1c397f273c382aff7920d7d97147d415f83/package.json#L81

Is there anything I can do to help? I suppose I can open a PR to update the peerDeps, but I'm not familiar with the project so I don't know if that fix is going to be "that simple"

**To Reproduce**

I can create a repo to reproduce this later today if that's helpful.

**Expected behavior**
I can install the library

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by alejandronanez**
Issue: Probably closes #99 

## What Changed

Add React18 to the peerDependency list

## Checklist

Check the ones applicable to your change:

- [ ] Tests are updated
- [ ] Documentation is updated

## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [x] `patch`
- [ ] `minor`
- [ ] `major`

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     "@storybook/client-api": ">=6.4.0",
     "@storybook/preview-web": ">=6.4.0",
     "@storybook/react": ">=6.4.0",
-    "react": "^16.8.0 || ^17.0.0"
```
Comment (ndelangen) [inline package.json:81]: Yeah this looks like the proper fix to me.

I'd expect a lock-file change associated with this, @yannbf
if for testing purposes you want to stay on a particular version in this repository (whilst support any listed version) it's a good idea to add the exact version you know this repo is ran with as a devDependency too.

Comment (alejandronanez) [reply package.json:81]: @ndelangen 👋 Just in case, I just ran `yarn install` locally and I didn't get any changes on my `yarn.lock` file.

**Comment by PeterSHL** [comment]
Hi @ndelangen do you know when is this expected to be released?

**Comment by ndelangen** [comment]
Yeah I'd assume so, @alejandronanez ?

**Pull request merged**